tests/datasets/test_streaming.py

```diff
@@ -0,0 +1,163 @@
+import math
+import os
+import shutil
+import tempfile
+import time
+from typing import Dict, Iterable, List, Optional
+
+import numpy as np
+import pytest
+import torch
+from torch.utils.data import DataLoader
+
+from composer.datasets.streaming import StreamingDataset, StreamingDatasetWriter
+
+
+def get_fake_samples_decoders(num_samples: int) -> List[Dict[str, bytes]]:
+    samples = [{"uid": f"{ix:06}".encode("utf-8"), "data": (3 * ix).to_bytes(4, "big")} for ix in range(num_samples)]
+    decoders = {
+        "uid": lambda uid_bytes: uid_bytes.decode("utf-8"),
+        "data": lambda data_bytes: int.from_bytes(data_bytes, "big")
+    }
+    return samples, decoders
+
+
+def write_synthetic_streaming_dataset(samples: List[Dict[str, bytes]], shard_size_limit: int) -> str:
+    tmpdir = tempfile.TemporaryDirectory().name
```

Comment:
For tests, can you use the `tmp_path` fixture that pytest provides? See https://docs.pytest.org/en/6.2.x/tmpdir.html

Reply:
Done!